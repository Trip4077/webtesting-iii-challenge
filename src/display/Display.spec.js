import React from 'react';
import { render, cleanup } from 'react-testing-library';

import pine from 'react-test-renderer';
import Display from './Display';

afterEach(cleanup);

describe('<Display />', () => {
    it('renders', () => {
        render(<Display />);
    });

    it('displays "Closed" if gate is closed', () => {
        const { getByText } = render(<Display closed={true} />);

        getByText(/closed/i);
    });

    it('displays "Open" if gate is not closed', () => {
        const { getByText } = render(<Display closed={false} />);

        getByText(/open/i);
    });

    it('displays "Locked" if gate is locked', () => {
        const { getByText } = render(<Display locked={true} />);

        getByText(/locked/i);
    });

    it('displays "Unlocked" if gate is not locked', () => {
        const { getByText } = render(<Display locked={false} />);

        getByText(/unlocked/i);
    });

    it('should have "red-led" class if gate is locked', () => {
        const { getByText } = render(<Display locked={true} />);

        const container = getByText(/locked/i);

        expect(container.classList.contains('red-led')).toBe(true);
    });

    it('should have "green-led" class if gate is unlocked', () => {
        const { getByText } = render(<Display locked={false} />);

        const container = getByText(/unlocked/i);

        expect(container.classList.contains('green-led')).toBe(true);
    });

    it('should have "red-led" class if gate is closed', () => {
        const { getByText } = render(<Display closed={true} />);

        const container = getByText(/closed/i);

        expect(container.classList.contains('red-led')).toBe(true);
    });

    it('matches snapshot: L - false, C - false', () => {
        const tree = pine.create(<Display />);

        expect(tree.toJSON()).toMatchSnapshot();
    });

    it('matches snapshot: L - true, C - false', () => {
        const tree = pine.create(<Display locked={true}/>);

        expect(tree.toJSON()).toMatchSnapshot();
    });

    it('matches snapshot: L - false, C - true', () => {
        const tree = pine.create(<Display closed={true} />);

        expect(tree.toJSON()).toMatchSnapshot();
    });

    it('matches snapshot: L - true, C - true', () => {
        const tree = pine.create(<Display locked={true} closed={true}/>);

        expect(tree.toJSON()).toMatchSnapshot();
    });
});