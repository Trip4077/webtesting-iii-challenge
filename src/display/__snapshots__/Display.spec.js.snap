// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Display /> matches snapshot: L - false, C - false 1`] = `
<div
  className="display panel"
>
  <div
    className="led green-led"
  >
    Unlocked
  </div>
  <div
    className="led green-led"
  >
    Open
  </div>
</div>
`;

exports[`<Display /> matches snapshot: L - false, C - true 1`] = `
<div
  className="display panel"
>
  <div
    className="led green-led"
  >
    Unlocked
  </div>
  <div
    className="led red-led"
  >
    Closed
  </div>
</div>
`;

exports[`<Display /> matches snapshot: L - true, C - false 1`] = `
<div
  className="display panel"
>
  <div
    className="led red-led"
  >
    Locked
  </div>
  <div
    className="led green-led"
  >
    Open
  </div>
</div>
`;

exports[`<Display /> matches snapshot: L - true, C - true 1`] = `
<div
  className="display panel"
>
  <div
    className="led red-led"
  >
    Locked
  </div>
  <div
    className="led red-led"
  >
    Closed
  </div>
</div>
`;
